extensions/data-workspace/src/dataworkspace.d.ts

```diff
@@ -115,4 +120,29 @@ declare module 'dataworkspace' {
 		 */
 		element: any;
 	}
+
+	export interface IProjectAction {
+		/**
+		 * id and display name of the project action
+		 */
+		readonly id: string;
+
+		/**
+		 * icon path of the project action
+		 */
+		readonly icon?: string | vscode.Uri | { light: string | vscode.Uri, dark: string | vscode.Uri };
+
+		/**
+		 * Run context for each project action
+		 * @param treeItem The treeItem in a project's hierarchy, to be used to obtain a Project
+		 */
+		run(treeItem: WorkspaceTreeItem): Promise<any>;
+	}
+
+	/**
+	 * List of project actions that should be grouped and have a separator after the last action
+	 */
+	export interface IProjectActionGroup {
+		actions: IProjectAction[];
```

Comment:
You're not allowing nested groups?

Reply:
Nested groups are allowed.
For the projectActions property, it can either be an array of actions or action groups or both
[Line 76] readonly projectActions: (IProjectAction | IProjectActionGroup)[];

Reply:
A nested group would be a group that contains other groups. Currently you don't have that being possible.

Reply:
Understood. As of now, I don't see a reason why we might need a nested group. 
The concept of groups came into picture so that icons could be divided into groups and can get a toolbar separator at the end.